Libraries/LibJS/AST.h

```diff
@@ -812,7 +816,7 @@ class ObjectProperty final : public ASTNode {
     virtual const char* class_name() const override { return "ObjectProperty"; }
 
     NonnullRefPtr<Expression> m_key;
-    NonnullRefPtr<Expression> m_value;
+    RefPtr<Expression> m_value;
```

Comment:
Why this change? My understanding is that `ObjectProperty` must always have a `value`.

Reply:
Not with a spread expression, which only has one value: `({ ...foo })`

Reply:
Other parsers have a separate class for that (something like `SpreadProperty`), maybe that would make sense for us too?

Reply:
Well it's already differentiated with a `Type` value, which is checked when executing the `ObjectExpression`. Seems like it would just add complexity since then it couldn't be a `Vector<ObjectExpression>`, we'd have to make a base class, and the execute method would have to do a bunch of casting. Just my two cents though.